solver/src/http_solver.rs

```diff
@@ -32,15 +32,15 @@ impl SolverConfig {
 pub struct HttpSolver {
     base: Url,
     client: Client,
-    api_key: Option<String>,
+    api_pwd: Option<&'static str>,
```

Comment:
I think we are going to get the base url as a command line argument. Then we can directly encode the username and password there (https://serverfault.com/questions/371907/can-you-pass-user-pass-for-http-basic-authentication-in-url-parameters) so there is no need to manually set username and password here.

Reply:
I would prefer my method, as the api_pwd will come from a secret env, instead of a normal parameter visible to anyone. But since this is security critical, I don't have a strong opinion

Reply:
Maybe we can setup the solver so that it is only accessible from within our kubernetes cluster. Even if not I don't feel this is really secret like the private keys. It's not a problem if we as the devs know the url with username and password.

Reply:
No, I think it is very convenient to test the solver also from our home computers. Then we don't have to setup the solver code.

But okay, then we will display the pwd in the kubernetes config file. That is not really an issue.